autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -835,19 +843,17 @@ def file_output(
 
         # Abort if we don't want to output anything.
         if hasattr(self, 'disable_file_output'):
-            if self.disable_file_output:
+            if 'all' in self.disable_file_output:
                 return None, {}
-            else:
-                self.disabled_file_outputs = []
 
         # This file can be written independently of the others down bellow
-        if 'y_optimization' not in self.disabled_file_outputs:
+        if 'y_optimization' not in self.disable_file_output:
             if self.output_y_hat_optimization:
                 self.backend.save_targets_ensemble(self.Y_optimization)
```

Comment:
as it says here, 
```
Whether this worker should output the target predictions, so that they are
stored on disk. Fundamentally, the resampling strategy might shuffle the
Y_train targets, so we store the split in order to re-use them for ensemble
selection.
```
It seems to me that our code for building an ensemble wont work if we don't run `self.backend.save_targets_ensemble(self.Y_optimization)`. So I guess we should remove this condition. Before doing that, as this code was used from `autosklearn`, @eddiebergman do you know the reason for this `self.output_y_hat_optimization` flag?

Reply:
We have this flag so that there is full control over what get's stored for memory purposes. We have a slighlty outdated PR that needs to be updated at somepoint but this is also all stored in memory during calculation which could be optimized.

Essentially, possibility to save memory if required

Reply:
but if you put `disable_file_output=['y_optimization`] the ensemble building fails and instead uses the single best model.

Reply:
Yup, but it's an option ... don't ask why ... I think the correct thing to do would be to check if is ensemble building is enabled and also check if `y_optimization` is included in `disable_file_output`.

Reply:
so you mean throw an error in this case?

Reply:
Yeah, or a warning, whatever you think is best.